### PR TITLE
converting kms as fixture

### PR DESCRIPTION
**Pull request by sandrampeter**
## Background

[Asana ](https://app.asana.com/0/1200681690870921/1201499983213189/f)

- KMS is now a fixture
- EC2 iam role is given access to build-specific KMS key and only root has access to KMS key policy. This is a best practice so that all accesses are operated and maintained in the IAM role policy
- KMS policy is updated to give TFE-S3 user access to CMK as ci test : scenario_aws_active_active_rhel7_proxy uses this user as object_storage_iam_user
- Output is utilized by other modules in root main.tf for encryption of other resources like redis, db, s3 etc
- All test now consume the CMK key created via fixture
- The postgres database is now encrypted at rest using CMK key. Since encryption at rest was enabled and no KMS key was provided, AWS Managed default key was used.

### CODE REVIEW

**Comment by anniehedgpeth** [comment]
@sandrampeter , since the ptfe-replicated tests were touched, would you mind testing those with Circle, too? Lmk if you need help with that.

**Comment by sandrampeter** [comment]
> @sandrampeter , since the ptfe-replicated tests were touched, would you mind testing those with Circle, too? Lmk if you need help with that.

test results : 
https://app.circleci.com/pipelines/github/hashicorp/ptfe-replicated/2919/workflows/456ba877-d9e5-43e9-8235-1ec9c993e7f6

File: fixtures/kms/main.tf [hunk 1]
```diff
@@ -11,8 +11,12 @@ resource "aws_kms_alias" "main" {
   target_key_id = aws_kms_key.main.key_id
 }
 
+data "aws_iam_user" "ci_s3" {
+  user_name = "TFE-S3"
+}
+
 resource "aws_kms_grant" "main" {
-  grantee_principal = var.iam_principal.arn
+  grantee_principal = data.aws_iam_user.ci_s3.arn
```
Comment (anniehedgpeth) [inline fixtures/kms/main.tf:19]: These fixture modules, unlike the test modules, are expected to be used by anyone. So the `grantee_principal` value here should be a variable, and the data block for the `aws_iam_user` should be moved to the tests that use that user.

**Comment by sandrampeter** [comment]
Updated CI test : https://app.circleci.com/pipelines/github/hashicorp/ptfe-replicated/2919/workflows/aa2746d4-068b-46e5-bece-a08c874bcd87

**Review by anniehedgpeth — APPROVED** [review]
lgtm!

**Pull request merged**